lib/Browser.php

```
@@ -35,10 +37,23 @@ class Browser implements BuzzClientInterface
     /** @var ResponseInterface */
     private $lastResponse;
 
-    public function __construct(BuzzClientInterface $client = null)
-    {
-        $this->client = $client ?: new FileGetContents();
-        $this->factory = new MessageFactory();
+    /**
+     * Browser constructor.
+     *
+     * @param BuzzClientInterface|null                      $client
+     * @param RequestFactoryInterface|RequestFactory|null   $requestFactory
+     * @param ResponseFactoryInterface|ResponseFactory|null $responseFactory To change the default response factory for FileGetContents
+     */
+    public function __construct(
+        BuzzClientInterface $client = null,
+        $requestFactory = null,
+        $responseFactory = null
+    ) {
+        $this->client = $client ?: new FileGetContents([], $responseFactory ?: new MessageFactory());
+        if (!$requestFactory instanceof RequestFactoryInterface && !$requestFactory instanceof RequestFactory) {
+            $requestFactory = new MessageFactory();
```

Rather throw an exception then silently correct the mistake.